lib/Sema/TypeCheckCodeCompletion.cpp

```diff
@@ -1303,25 +1303,28 @@ void KeyPathTypeCheckCompletionCallback::sawSolution(
       // to look up type variables by their locators.
       auto RootLocator =
           S.getConstraintLocator(KeyPath, {ConstraintLocator::KeyPathRoot});
-      auto BaseVariableType =
+      auto BaseVariableTypeBinding =
```

Comment:
This is very strange, the only reason why root is not in the bindings is key path isn't in scope of the solution since solution always records all of the type variables present in the constraint system...

Reply:
My gut feeling was that it’s because the key path was skipped due to invalid code inside result builders. Do you think that might be the cause?

Reply:
Hm, maybe if part of the statement is skipped...